### PR TITLE
cryptsetup: Lock whole block device before attach

### PR DESCRIPTION
**Pull request by daandemeyer**
Let's avoid udev and systemd-crypsetup messing with a block device concurrently.

### CODE REVIEW

File: src/cryptsetup/cryptsetup.c [hunk 1]
```diff
@@ -13,13 +13,15 @@
...
 #include "ask-password-api.h"
+#include "blockdev-util.h"
 #include "build.h"
 #include "cryptsetup-fido2.h"
 #include "cryptsetup-keyfile.h"
 #include "cryptsetup-pkcs11.h"
 #include "cryptsetup-tpm2.h"
 #include "cryptsetup-util.h"
 #include "device-util.h"
+#include "fd-util.h"
```
Comment (YHNdnzj) [inline src/cryptsetup/cryptsetup.c]: Not sorted

**Comment by poettering** [comment]
hmm, why does this matter? can you explain what problem you ran into?

generally the assumption is that you take the lock when you change the fs metadata, i.e. reformat, wipe, create partitions, remove them and so on. But if you just unlock a crypt sevice, or mount an fs without changing the general structure of it, the locking should not be necessary.

or what am I missing?

**Comment by DaanDeMeyer** [comment]
@poettering This failure in TEST-70-TPM2 which goes away with this change: https://paste.centos.org/view/50e40bde

**Comment by DaanDeMeyer** [comment]
Actually nvm I found another run failing with the same issue even with this commit.

**Pull request closed without merging**